entrypoint.sh

```diff
@@ -9,20 +26,30 @@ FILES=$(curl -s -X GET -G $URL | jq -r '.[] | select( (.filename | endswith("man
 rc=0
 SUCCESS_TARGETS=()
 FAILED_TARGETS=()
-if [ -z "$FILES" ]
-then
-      echo "nothing to build"
+if [ -z "$FILES" ]; then
+    echo "nothing to build"
 else
-      echo "targets to build: $FILES"
-      for TAR_PATH in $FILES
-      do
-        if jina hub build $TAR_PATH --pull --prune-images --test-uses --raise-error --daemon; then
+    echo "targets to build: $FILES"
+    for TAR_PATH in $FILES; do
+      cmd="jina hub build --pull --prune-images"
+      if [[ "$IS_PUSH" == true ]]; then
+        cmd="$cmd --push $TAR_PATH"
+
+        # only add args when not empty
+        [ ! -z "$DOCKERHUB_USERNAME" ] && cmd="$cmd --username $DOCKERHUB_USERNAME"
+        [ ! -z "$DOCKERHUB_PASSWORD" ] && cmd="$cmd --password $DOCKERHUB_PASSWORD"
+        [ ! -z "$DOCKERHUB_REGISTRY" ] && cmd="$cmd --registry $DOCKERHUB_REGISTRY"
```

Comment:
Since we are not using `getopts` and reading bash arguments with `$1, $2 ..` etc., the script will fail is we don't have 9 arguments, which means, `$DOCKERHUB_USERNAME` cannot be empty. We can rather move to `getopts` for better handling

Reply:
it always has 9 args as written in `action.yml`, as the args fed to `entrypoint.sh` are controlled by Github action directly.